x/eibc/types/demand_order.go

```diff
@@ -89,6 +98,18 @@ func (m *DemandOrder) GetRecipientBech32Address() sdk.AccAddress {
 	return recipientBech32
 }
 
+func (m *DemandOrder) IsActive() error {
+	// Check that the order is not fulfilled yet
+	if m.IsFulfilled {
+		return ErrDemandAlreadyFulfilled
+	}
+	// Check the underlying packet is still relevant (i.e not expired, rejected, reverted)
+	if m.TrackingPacketStatus != commontypes.Status_PENDING {
+		return ErrDemandOrderInactive
+	}
+	return nil
+}
```

Comment:
Can we have any of

```
Open order
Unexecuted order
Outstanding order
Live order
```

as terminology instead of active

Also, bit weird to return errors here

They aren't really errors, they're just statuses.

Just make an enum

Reply:
will rename
It's not that wierd, as we expect to interact with only "Outstanding" orders.
maybe `ValidateLiveOrder` instead of `IsActive` is more suitable